### PR TITLE
admin/info: Add HTTPStats value as part of serverInfo() struct.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Remove our counter implementation instead use atomic external
package which supports more types and methods.

## Motivation and Context
Add HTTPStats value as part of serverInfo() struct.

## How Has This Been Tested?
Manually.. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/generic-handlers.go [hunk 1]
```diff
@@ -401,9 +401,18 @@ func (h httpStatsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
 
+	// Time after call has completed.
+	tAfter := UTCNow()
+
+	// Time duration since the call started.
+	duration := tAfter.Sub(tBefore)
+
 	// Update http statistics
-	globalHTTPStats.updateStats(r, ww)
+	globalHTTPStats.updateStats(r, ww, duration.Seconds())
```
Comment (vadmeste) [inline cmd/generic-handlers.go]: Can you add a comment saying that `duration.Seconds()` returns a floating and non rounded seconds ?

File: cmd/http-stats.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+	totalDELETEs   HTTPMethodStats
+	successDELETEs HTTPMethodStats
+}
+
+// Converts http stats into struct to be sent back to the client.
+func (st HTTPStats) toServerHTTPStats() ServerHTTPStats {
+	serverStats := ServerHTTPStats{}
+	serverStats.TotalHEADStats = ServerHTTPMethodStats{
+		Count:       st.totalHEADs.Counter.Load(),
+		AvgDuration: fmt.Sprint(time.Duration(st.totalHEADs.Duration.Load()/float64(st.totalHEADs.Counter.Load())) * time.Second),
```
Comment (vadmeste) [inline cmd/http-stats.go]: Can you avoid duplicate fmt.Sprint()? It would be easier to review.

**Review by vadmeste — APPROVED** [review]
LGTM, only minor comments.

File: cmd/generic-handlers.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/generic-handlers.go]: Done

File: cmd/http-stats.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/http-stats.go]: Done

**Review by vadmeste — APPROVED** [review]

**Comment by harshavardhana** [comment]
ping @krisis - can you review.

**Comment by harshavardhana** [comment]
Since @krisis is not around @donatello can you review.

**Review by krisis — APPROVED** [review]

File: cmd/generic-handlers.go [hunk 3]
```diff
@@ -401,9 +401,21 @@ func (h httpStatsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
 	h.handler.ServeHTTP(ww, r)
 
+	// Time after call has completed.
+	tAfter := UTCNow()
+
+	// Time duration in secs since the call started.
+	//
+	// We don't need to do nanosecond precision in this
+	// simply for the fact that it is not human readable.
+	durationSecs := tAfter.Sub(tBefore).Seconds()
```
Comment (mtparet) [inline cmd/generic-handlers.go]: Does it means that duration below 1s are ignored ?

**Pull request merged**